client/src/app/core/repositories/motions/motion-repository.service.ts

```diff
@@ -692,37 +705,41 @@ export class MotionRepositoryService extends BaseIsAgendaItemAndListOfSpeakersCo
         const motion = this.getAmendmentBaseMotion(amendment);
         const baseParagraphs = this.getTextParagraphs(motion, true, lineLength);
 
-        return amendment.amendment_paragraphs
-            .map(
-                (newText: string, paraNo: number): DiffLinesInParagraph => {
-                    if (newText !== null) {
-                        return this.diff.getAmendmentParagraphsLinesByMode(
-                            paraNo,
-                            baseParagraphs[paraNo],
-                            newText,
-                            lineLength
-                        );
-                    } else {
-                        // Nothing has changed in this paragraph
-                        if (includeUnchanged) {
-                            const paragraph_line_range = this.lineNumbering.getLineNumberRange(baseParagraphs[paraNo]);
-                            return {
-                                paragraphNo: paraNo,
-                                paragraphLineFrom: paragraph_line_range.from,
-                                paragraphLineTo: paragraph_line_range.to,
-                                diffLineFrom: paragraph_line_range.to,
-                                diffLineTo: paragraph_line_range.to,
-                                textPre: baseParagraphs[paraNo],
-                                text: '',
-                                textPost: ''
-                            } as DiffLinesInParagraph;
+        if (baseParagraphs && baseParagraphs.length) {
```

Comment:
Now, the return type is not correct anymore. If you adjust the return type, add null checks everywhere this function is called.